app/actions/JoblistingActions.js

```diff
@@ -63,47 +55,31 @@ export function createJoblisting({
   fromYear,
   toYear,
   applicationUrl
-}: Object): Thunk<*> {
-  return dispatch => {
-    dispatch(startSubmit('joblistingEditor'));
-
-    return dispatch(
-      callAPI({
-        types: Joblistings.CREATE,
-        endpoint: '/joblistings/',
-        method: 'POST',
-        body: {
-          title,
-          text,
-          company: company && company.value,
-          responsible: responsible.value,
-          description,
-          deadline: moment(deadline).toISOString(),
-          visibleFrom: moment(visibleFrom).toISOString(),
-          visibleTo: moment(visibleTo).toISOString(),
-          jobType,
-          workplaces,
-          fromYear,
-          toYear,
-          applicationUrl
-        },
-        schema: joblistingsSchema,
-        meta: {
-          errorMessage: 'Opprettelse av jobbannonse feilet'
-        }
-      })
-    )
-      .then(action => {
-        if (!action || !action.payload) return;
-        const id = action.payload.result;
-        dispatch(stopSubmit('joblistingEditor'));
-        dispatch(push(`/joblistings/${id}`));
-      })
-      .catch(action => {
-        const errors = { ...action.error.response.jsonData };
-        dispatch(stopSubmit('joblistingEditor', errors));
-      });
-  };
+}: Object) {
+  return callAPI({
+    types: Joblistings.CREATE,
+    endpoint: '/joblistings/',
+    method: 'POST',
+    body: {
```

Comment:
why does destructure its argument? would look at lot simpler if it was something like:

```js
export function createJoblisting(data: Object) {
  return callAPI({
    types: Joblistings.CREATE,
    endpoint: '/joblistings/',
    method: 'POST',
    body: {
      ...data,
      company: data.company && data.company.value,
      responsible: data.responsible.value,
      deadline: moment(data.deadline).toISOString(),
      visibleFrom: moment(data.visibleFrom).toISOString(),
      visibleTo: moment(data.visibleTo).toISOString(),
    },
    schema: joblistingsSchema,
    meta: {
      errorMessage: 'Opprettelse av jobbannonse feilet'
    }
  });
}
```

the moment stuff and such could be put in a function that could be shared with the edit thing as well

Reply:
```jsx
export function createJoblisting({company, responsible, deadline, visibleFrom, visibleTo, ...data}: Object) {
```
Much less typing, yeah..